### DISCUSSION TITLE
Why use this? (Read description also)

### DISCUSSION THREAD

**Discussion by lotus647** [coder/code-server#6329] [Q&A] [answered]
Hello everyone,

I don't wanna make the creator feel bad about their skills, and the time and effort they put into this (probably alot) but why should I use this? (Keep reading for explanation)

There's already a web version of VSCode, located at [https://vscode.dev](https://vscode.dev). So what's the point? Or, you could just use the desktop version.

If it is just for personal experience, or contributing to the community, thats an adequate answer, but im just wondering if there was any monetary gain or sum.

Cheers, 

theClawsmos (pls no hate for this question)

**Comment by code-asher** [comment, marked as answer]
https://vscode.dev has no backend which makes it fairly useless in 
many cases but if you are referring to the web version that also 
has a backend (which is available from the VS Code GitHub 
repository), then the differences between that and code-server are 
minor.  The gist is that code-server wraps VS Code to make it more 
friendly to self-hosting.  You can read more about those 
differences here:
https://github.com/coder/code-server/discussions/5336#discussioncomment-4925471

**Reply by lotus647** [reply]
Cool okay

**Reply by lg-nineoneninethreenine** [reply]
Well, just was gonna say that and a whole lot more. You could also try comparing Coder's code-server with Microsoft's code-server, and lemme tell ya where Microsoft loses. I tried it on my 4GB computer, and the thing barely started, whereas code-server did. In my opinion, code-server perfected the Web IDE idea. You could just use it on your browser, or if you are a genius, you could do something like nginx, Caddy and domains to actually host it and access it from any computer(I distinctly remember doing this on Railway, it was very fun).

**Reply by lotus647** [reply]
um what? very confused cuz you jump around on topics lmao

**Reply by lg-nineoneninethreenine** [reply]
Sorry, I think my mind was running fast at that time. What I mean to say is that Microsoft has their own backend with vscode.dev, and they call it code-server too. But it can only run on servers since when I tested it, it barely started, which just cleanly shows the best advantage of code-server - lightweight to use. Usually it runs on localhost, but if you really know the things with nginx, Caddy and more, you could expose the port and link it to a domain. Now you can code from any device, anywhere in the world. This was done with Docker images, on a platform called Railway. I tried it, and while I just kept it running, it did the job.

Hope you understood now, I guess I was just thinking unusually fast. Maybe 'cause it was midnight for me. 🤷🏻‍♂️

**Reply by lotus647** [reply]
yes. okay much clearer. 💯

**Comment by benz0li** [comment]
> The gist is that code-server wraps VS Code to make it more friendly to self-hosting.

@theClawsmos This allows creating something like @b-data's JupyterLab docker stacks, which contain code-server next to [JupyterLab](https://github.com/jupyterlab/jupyterlab):

* [JupyterLab R docker stack](https://github.com/b-data/jupyterlab-r-docker-stack)
* [JupyterLab Python docker stack](https://github.com/b-data/jupyterlab-python-docker-stack)
* [JupyterLab Julia docker stack](https://github.com/b-data/jupyterlab-julia-docker-stack)

ℹ️ From a Data Science perspective: The beauty of 'VS Code'/code-server is that it has extensions for [R](https://open-vsx.org/extension/REditorSupport/r), [Python](https://open-vsx.org/extension/ms-python/python) and [Julia](https://open-vsx.org/extension/julialang/language-julia) – today's major Data Science languages.
